# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build libpng-build tiff-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

pkg-config --exists lcms2-build && REQUIRES="$REQUIRES lcms2-build"

# configure (ac-2.67; am-1.11; lt-2.2.6b)
libtoolize -c -f; aclocal -I m4; autoheader; autoconf; automake -a -c
./configure --disable-static --prefix=

# build
make

# install
install -d $DESTDIR/bin
install applications/codec/.libs/image_to_j2k $DESTDIR/bin
install applications/codec/.libs/j2k_dump $DESTDIR/bin
install applications/codec/.libs/j2k_to_image $DESTDIR/bin
install -d $DESTDIR/include/openjpeg-1.5
install -m 644 libopenjpeg/openjpeg.h $DESTDIR/include/openjpeg-1.5
ln -sf openjpeg-1.5/openjpeg.h $DESTDIR/include/openjpeg.h
install -d $DESTDIR/lib
install libopenjpeg/.libs/libopenjpeg.so.1.* $DESTDIR/lib/libopenjpeg.so.1
ln -sf libopenjpeg.so.1 $DESTDIR/lib/libopenjpeg.so
install -d $DESTDIR/lib/pkgconfig
install -m 644 libopenjpeg1.pc $DESTDIR/lib/pkgconfig
ln -sf libopenjpeg1.pc $DESTDIR/lib/pkgconfig/libopenjpeg.pc

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/openjpeg-build.pc << EOF
Name: openjpeg
Version: 1.5.1
Description: JPEG 2000 library
Requires: $REQUIRES

devel=\\
/include/openjpeg.h \\
/include/openjpeg-1.5/openjpeg.h \\
/lib/libopenjpeg.so \\
/lib/pkgconfig/libopenjpeg.pc \\
/lib/pkgconfig/libopenjpeg1.pc

exec=\\
/bin/image_to_j2k \\
/bin/j2k_dump \\
/bin/j2k_to_image \\
/lib/libopenjpeg.so.1
EOF

exit

# patch
--- openjpeg-1.5.1.orig/configure.ac
+++ openjpeg-1.5.1/configure.ac
@@ -29,6 +29,8 @@
 AM_INIT_AUTOMAKE([1.11 foreign dist-bzip2 dist-xz dist-zip])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 
+AM_MAINTAINER_MODE
+
 MAJOR_NR=OPJ_MAJOR
 MINOR_NR=OPJ_MINOR
 MICRO_NR=OPJ_MICRO

# source
http://downloads.sourceforge.net/openjpeg.mirror/openjpeg-1.5.1.tar.gz
